### DISCUSSION TITLE
Module hook itself

### DISCUSSION THREAD

**Discussion by Karanveer7921** [LSPosed/LSPosed#1961] [Q&A] [answered]
I have an android app + xposed module. In the app, I have to show wether xposed module is enabled or not. Normally, what we can do is create a method in app class like, isXposedModuleEnabled() and set it to return false. Via xposed api, hook this method and return true. I dont only need to show xposed status, but also need to hook android internal api which my app uses. so basically need to hook the module app itself

**Comment by yujincheng08** [comment, marked as answer]
That's meaningless.

**Reply by Karanveer7921** [reply]
assume there is a game and via xposed i provide some tweaks for that game. There might be switches to turn on and off hooks, i could also display target game's logs in my app (xposed module) via hooking the logger for that game or something like that. what i want is if xposed module isnt activated, it should show a screen to user that module isnt activated and don't allow user to do anything with those switches/options. If you look at the source of other modules which shows the status of if this module is activated or not, they used the same approach (creating a method that return constant false, then hook it to return true, which shows module is activated) but this wont work in lsposed or edxposed since they dont allow to add module itself in the scope list

**Reply by yujincheng08** [reply]
the module itself will be in the scope list if there's at least one other app hooked

**Reply by Karanveer7921** [reply]
It never invoked the IXposedHookLoadPackage#handleLoadPackage(...) for the module itself , however It invokes for other apps in the scope

**Reply by yujincheng08** [reply]
It will invoke. My module uses the same way to check the activation status.

**Reply by Karanveer7921** [reply]
Okay it invoked now, I reinstalled the module app and rebooted the device

**Comment by yujincheng08** [comment]
If you want to hook some system methods to do some tricks or to measure the performance, why not https://github.com/LSPosed/LSPlant
